spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkAppenderFactory.java

```diff
@@ -161,7 +162,12 @@ private StructType lazyPosDeleteSparkType() {
   }
 
   @Override
-  public FileAppender<InternalRow> newAppender(OutputFile file, FileFormat fileFormat) {
+  public FileAppender<InternalRow> newAppender(OutputFile outputFile, FileFormat format) {
+    return newAppender(EncryptionUtil.plainAsEncryptedOutput(outputFile), format);
+  }
+
+  @Override
+  public FileAppender<InternalRow> newAppender(EncryptedOutputFile file, FileFormat fileFormat) {
```

Comment:
Ah, I was expecting these changes in Spark 3.5. Feel free to move them there, or we can add them to Spark 3.4 and open follow up PRs.

Reply:
Sounds good, I'll move them to 3.5.